### DISCUSSION TITLE
Why does scheduler_job set TI state to NONE?

### DISCUSSION THREAD

**Discussion by darinbaumgartel** [apache/airflow#25620] [General]
Trying to understand this line: https://github.com/apache/airflow/blob/3dbca5e3f224dd8b49fc91459a368eb30e5b69a7/airflow/jobs/scheduler_job.py#L537 

What is the scenario where the dagrun is finished and the TI state needs to be set to NONE? Shouldn't all tasks being examined in this list be in a scheduled state? 

Thank you!

**Comment by potiuk** [comment]
I guess it is to handle some race condition. But It's notoriously difficult to get all complex scenarios. And I think knowledge of anyne here is the same as you - looke at the code and try to figure it out, test and fix if you think it is wrong., 

If you think it is a bug and can reproduce it, write a test and provide a PR fixing it, that woudl be awesome.
